### ISSUE TITLE
Support for simple open/close contacts (Doors/Windows)

### ISSUE THREAD

**Issue by catsoo** [openhab/openhab-google-assistant#160]
It would be great to also support basic Doors/Windows with an Open/Close state

`Number/String { ga="Window" }`
`Number/String { ga="Door" }`

In terms of Device Attributes (GA Side) in Devices.js

`discreteOnlyOpenClose=true`
`queryOnlyOpenClose =true`

On the OpenHab side items would use states "OPEN/CLOSED" (HomematicIP contacts for example) or 0/1 in case of Numbers.

**Comment by michikrug** [comment]
Hi. Thanks for raising the request.

Why is using the Rollershutter item type not an option? It also has 0/100 values as well as open/close states.

**Comment by michikrug** [comment]
Adding the attributes is not a problem and I will have a closer look there.

**Comment by catsoo** [comment]
In general because channels are usually String/Number types, so you'd need to create additional proxy items and rules to map states (OPEN -> 100, CLOSED -> 0) to a rollershutter.

**Comment by michikrug** [comment]
Can you please help me with a real world example to understand the requirements better?

**Comment by catsoo** [comment]
A real world example would be this:

`String wz_kontakt_tuer          "Wohnzimmertür"             (gContacts,gWZ) {channel="homematic:XXXX#STATE"}
`
possible item values are _OPEN_ or _CLOSED_

**Comment by smarthome-w** [comment]
Please add also contact style support to lock:
```
Contact my_lock "Family room lock" (gLocks) {channel="homematic:XXXX#STATE", ga="Lock"}
```
with the standard question: "Is family room door locked"?

For now lock is also switch oriented and there is no ability to get lock (window, etc.) status. Contact is very useful because it is read-only in user interface and doesn't generate unnecessary events and state changes.

**Comment by michikrug** [comment]
I implemented support for Contact items.
PR will be opened soon.

**Comment by SeveranExp** [comment]
Is this available already? Can you point me to documentation? (Bit new on this side, sorry if it's too basic of a question)

**Comment by michikrug** [comment]
Unfortunately, this is not yet released to the public :( See #165

There is also some documentation added.

**Comment by catsoo** [comment]
Closing - support was merged with https://github.com/openhab/openhab-google-assistant/pull/175

String item support not needed from my side anymore, as this issue was addressed with OH3x where all of my bindings now also support contact items.
